### ISSUE TITLE
sshguard doesn't start ("No such file or directory")

### ISSUE THREAD

**Issue by Criena** [NixOS/nixpkgs#33313]
## Issue description

It seems that there is something wrong with sshguard. I enabled it, no additional configuration. But it doesn't start.

The log shows the following:

> Jan 02 07:26:11 HOSTNAME systemd[1076]: sshguard.service: Failed at step NAMESPACE spawning /nix/store/40abbrf8bybnj8bsf1xrzr2f771ggr39-sshguard-2.0.0/bin/sshguard: **No such file or directory**
> Jan 02 07:26:11 HOSTNAME systemd[1]: sshguard.service: Main process exited, code=exited, status=226/NAMESPACE
> Jan 02 07:26:11 HOSTNAME systemd[1077]: sshguard.service: Failed at step NAMESPACE spawning /nix/store/i8f2vk393pz3a445vjvvzkgca78qa6c4-unit-script/bin/sshguard-post-start: **No such file or directory**

Which is very peculiar, since both paths do exist and are marked executable:

> -r-xr-xr-x 2 root root 2.2K Jan  1  1970 /nix/store/40abbrf8bybnj8bsf1xrzr2f771ggr39-sshguard-2.0.0/bin/sshguard
> -r-xr-xr-x 2 root root 578 Jan  1  1970 /nix/store/i8f2vk393pz3a445vjvvzkgca78qa6c4-unit-script/bin/sshguard-post-start

### Steps to reproduce

Include sshguard in configuration (services.sshguard.enable = true;) -- that's it.

## Technical details

> Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

[root@HOSTNAME:~]# nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.72, NixOS, 17.09.2594.af6aacddc18 (Hummingbird)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-17.09.2594.af6aacddc18"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by fruser** [comment]
having the same issue :(

**Comment by Criena** [comment]
After some investigation I found out that systemd fails to start the service due to the following setting in sshguard.service:

> ReadWriteDirectories=/run/sshguard /var/lib/sshguard

If these directories do not exist, systemd refuses to start the service. The solution would therefore to either create the directories during the installation or to add a prefix to the directories ("-"), which makes systemd ignore the non-existance of the directories.

As a quick workaround I created the two directories manually. Works fine now.
